lib/phoenix_live_view/utils.ex

```diff
@@ -284,21 +284,37 @@ defmodule Phoenix.LiveView.Utils do
     """
   end
 
+  @doc """
+  Returns a map of infos about the lifecycle stage for the given `view`.
+  """
+  def lifecycle(%Socket{} = socket, view, stage, arity) do
+    info = %{
+      callbacks?: Lifecycle.callbacks?(socket, stage),
+      exported?: function_exported?(view, stage, arity),
+      view: view
+    }
+
+    Map.put(info, :any?, info.callbacks? or info.exported?)
+  end
+
   @doc """
   Calls the `c:Phoenix.LiveView.mount/3` callback, otherwise returns the socket as is.
   """
   def maybe_call_live_view_mount!(%Socket{} = socket, view, params, session) do
-    {cont_or_halt, %Socket{} = socket} = Lifecycle.mount(params, session, socket)
+    lifecycle = lifecycle(socket, view, :mount, 3)
 
-    if function_exported?(view, :mount, 3) do
+    if lifecycle.any? do
       :telemetry.span(
         [:phoenix, :live_view, :mount],
         %{socket: socket, params: params, session: session},
         fn ->
           socket =
-            case cont_or_halt do
-              :cont -> view.mount(params, session, socket)
-              :halt -> {:ok, socket}
+            case {Lifecycle.mount(params, session, socket), lifecycle} do
+              {{:cont, %Socket{} = socket}, %{exported?: true}} ->
+                lifecycle.view.mount(params, session, socket)
```

Comment:
Someone can define a `on_mount` hook but not the `mount/3` function. This is exactly the same case as the `handle_params`, so we need to check if it is exported? or not right here. :)

Reply:
> need to check if it is exported?

L313 does that check, so `{:cont, socket}` + `exported?` invokes mount, all other cases fall through.

This is what I meant when I said it was over-complicated :)

Reply:
Ignore me, I missed the exported?: true pattern match above. My bad. :)